src/auto-evo/steps/ModifyExistingSpecies.cs

```diff
@@ -0,0 +1,337 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+using Mutation = System.Tuple<MicrobeSpecies, MicrobeSpecies, RunResults.NewSpeciesType>;
+
+public class ModifyExistingSpecies : IRunStep
+{
+    private readonly Patch patch;
+    private readonly SimulationCache cache;
+
+    private readonly WorldGenerationSettings worldSettings;
+
+    public ModifyExistingSpecies(Patch patch, SimulationCache cache, WorldGenerationSettings worldSettings)
+    {
+        this.patch = patch;
+        this.cache = cache;
+        this.worldSettings = worldSettings;
+    }
+
+    public int TotalSteps => 1;
+
+    public bool CanRunConcurrently => true;
+
+    /// <summary>
+    ///   Returns a new list of all possible species that might emerge in response to the provided pressures,
+    ///   as well as a copy of the original species.
+    /// </summary>
+    /// <returns>List of viable variants, and the provided species</returns>
+    public static List<MicrobeSpecies> GenerateMutations(MicrobeSpecies baseSpecies, int amount,
+        SimulationCache cache, List<SelectionPressure> selectionPressures, Random random)
+    {
+        // TODO: Possibly make that 100 a difficulty setting
```

Comment:
Doesn't the MP numbers here need to be hooked up to the AI mutation rate variable that is in new game settings? Otherwise that option will stop working.